### PR TITLE
Doc bootstrap ta

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Documentation for https://github.com/OP-TEE/optee_os/pull/1928

### CODE REVIEW

**Comment by jforissier** [comment]
LGTM

File: documentation/optee_design.md [hunk 1]
```diff
@@ -645,9 +645,19 @@ they execute in Secure World.
...
 
-#### "Normal" or REE FS Trusted Applications
+#### "Normal" or Secure Storage Trusted Applications
+
+These are stored in secure storage. The meta data is stored in a database
+of all installed TAs and the actual binary is stored encrypted as a
+separate file in the untrusted REE filesystem.
+
+Before these TAs can be loaded they have to be installed first, this is
+something that can be done during initial deployment or at a later stage.
```
Comment (jforissier) [inline documentation/optee_design.md]: Worth mentioning that `xtest --install-ta` may be used for that?

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Rebased, squashed and tag applied

File: documentation/optee_design.md [hunk 2]
```diff
@@ -609,8 +609,9 @@ to as 'Trusted Applications'. For most cases, user mode TAs are preferred.
 
 ### Pseudo Trusted Applications
 
-These are added directly to the OP-TEE core tree in, eg, `core/arch/arm/pta`,
-and are built along with and statically built into the OP-TEE core blob.
+These are implemented directly to the OP-TEE core tree in, eg,
+`core/arch/arm/pta`, and are built along with and statically built into the
```
Comment (etienne-lms) [inline documentation/optee_design.md]: (...) and are built along with and statically **~~built~~ linked** into the (...)

File: documentation/optee_design.md [hunk 3]
```diff
@@ -609,8 +609,9 @@ to as 'Trusted Applications'. For most cases, user mode TAs are preferred.
...
+OP-TEE core blob.
```
Comment (etienne-lms) [inline documentation/optee_design.md]: s/blob/firmware/ ?

Comment (jenswi-linaro) [reply documentation/optee_design.md]: I think the word firmware should be avoided to not be so easily confused with ARM Trusted Firmware.

File: documentation/optee_design.md (hunk 2 above)
Comment (jenswi-linaro) [reply documentation/optee_design.md]: This sentence isn't the best, but exchanging **built into** with **linked into** doesn't make it better. 
IMO, the words **linked into** implies some additional isolation compared to the other code.

**Comment by jenswi-linaro** [comment]
It seems @etienne-lms accepted my answers.
Tag applied.

**Comment by etienne-lms** [comment]
yes, sorry. sure....

**Pull request merged**